client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,110 @@
+	const swiperContainerId = `swiper-container-${ collectionSlug }`;
+
+	const [ isSwiperLoaded, setSwiperLoaded ] = useState( false );
+	useEffect( () => {
+		if ( ! isSwiperLoaded ) {
+			const el = document.querySelector( `#${ swiperContainerId }` ) as HTMLElement;
+			if ( el ) {
+				/**
+				 * We have to import the swiper package dynamically because it doesn't offer a CommonJS version. Because of this, the SSR build will fail.
+				 *
+				 * From a performance standpoint, it should not cause any issues; we are already loading the ThemeCards dynamically anyway.
+				 */
+				Promise.all( [
+					import( 'swiper' ),
+					import( 'swiper/modules' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/navigation' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/keyboard' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/mousewheel' ),
+				] ).then( ( values ) => {
+					const Swiper = values[ 0 ].default;
+					const { Navigation, Keyboard, Mousewheel } = values[ 1 ];
+
+					swiperInstance.current = new Swiper( el, {
+						cssMode: true,
+						mousewheel: true,
+						keyboard: {
+							enabled: true,
+							onlyInViewport: false,
+						},
+						navigation: {
+							nextEl: '.theme-collection__carousel-nav-button--next',
+							prevEl: '.theme-collection__carousel-nav-button--previous',
+						},
+						threshold: 5,
+						slideToClickedSlide: true,
+						slidesPerView: 'auto',
+						spaceBetween: 20,
+						modules: [ Navigation, Keyboard, Mousewheel ],
+					} );
+					setSwiperLoaded( true );
+				} );
+			}
+		}
+
+		return () => {
+			swiperInstance.current?.destroy();
+		};
+	}, [ swiperContainerId, isSwiperLoaded ] );
+
+	return (
+		<div className="theme-collection__container ">
+			<h2>{ heading }</h2>
+			{ subheading }
+			<div className="swiper-container" id={ swiperContainerId }>
+				<div className="theme-collection__carousel-controls">
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--previous">
+						<Icon icon={ chevronLeft } />
+					</Button>
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--next">
+						<Icon icon={ chevronRight } />
+					</Button>
+				</div>
+				<div className="theme-collection__list-wrapper swiper-wrapper">{ children }</div>
+			</div>
+		</div>
+	);
+}
+
+export function CollectionListItem( {
+	collectionSlug,
+	themeId,
+	children,
+}: PropsWithChildren< { collectionSlug: string; themeId: string } > ) {
+	return (
+		<div
+			key={ `theme-collection-container-${ collectionSlug }-${ themeId }` }
+			className="theme--collection__list-item swiper-slide"
```

Comment:
```suggestion
			className="theme-collection__list-item swiper-slide"
```

(see [CSS guidelines](https://github.com/Automattic/wp-calypso/blob/7752207da247005f4b24898bad1807e16a474d92/docs/coding-guidelines/css.md))